handlers.go

```diff
@@ -515,21 +516,30 @@ func (s *Server) SaveBlob(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	tf, err := os.OpenFile(path, os.O_CREATE|os.O_WRONLY|os.O_EXCL, 0600)
+	_, err = os.Stat(path)
+	if err == nil {
+		http.Error(w, http.StatusText(http.StatusForbidden), http.StatusForbidden)
+		return
+	}
+	if !os.IsNotExist(err) {
+		if s.Debug {
+			log.Print(err)
+		}
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	tf, err := ioutil.TempFile(filepath.Dir(path), ".rest-server-temp")
```

Comment:
Perhaps add the blob ID to the temp file template for debugging?

Reply:
I've add the blob id to the filename